solidity/contracts/Rewards.sol

```diff
@@ -469,9 +478,36 @@ contract Rewards {
         if (interval > allocatedIntervals) {
             allocateRewards(interval.sub(1));
         }
-        uint256 totalAllocation = _adjustedAllocation(interval);
-        unallocatedRewards = unallocatedRewards.sub(totalAllocation);
-        intervalAllocations.push(totalAllocation);
+        uint256 baseAllocation = _baseAllocation(interval);
+        uint256 adjustedAllocation = _adjustedAllocation(interval);
+        uint256 allocationDifference = baseAllocation.sub(adjustedAllocation);
+        intervalAllocations.push(adjustedAllocation);
+        // If a new rewards contract has been defined and the quota isn't met,
+        // send the excess to the new contract.
+        // Otherwise keep the excess unallocated.
+        if (newRewardsSpecified() && (allocationDifference > 0)) {
+            unallocatedRewards = unallocatedRewards.sub(baseAllocation);
+            sendToNewRewards(allocationDifference);
```

Comment:
The assumption here is: Stakers will migrate to the new operator contract and we will not meet the minimum groups quota here. Over time, we can move unallocated tokens to the new rewards contract.

This is very elegant but I am not sure if we should handcuff ourselves. There are two issues I see:
- The minimum group quota for random beacon is set to 2 (group expiration time is two weeks, interval is 30 days). Anyone with 64 * min_stake, can block the migration.
- It will take a lot of time until we reallocate tokens, even if no groups will be created. And the rewards curve on the new contract will not have the expected shape.

Instead, I propose to set one interval governance delay and once the last interval is allocated, move all the unallocated tokens out from this contract to the new one.

WDYT?

Reply:
An alternative proposition: https://github.com/keep-network/keep-core/pull/2108